hermes/hermes.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Evan Spendlove (@evanSpendlove)
+//
+// hermes.go defines the structures necessary for the main hermes object to
+// monitor a storage system.
+
+package hermes
+
+import (
+	"context"
+
+	"github.com/golang/glog"
+	"github.com/google/cloudprober"
+)
+
+// FileOperation is an int used as part of the FileOperation enum within the intent log of Hermes' StateJournal.
+type FileOperation int
+
+// FileOperation enum is used for marking the operation intent within the StateJournal of Hermes.
+// FileOperation has two possible values: CREATE, DELETE.
+const (
+	Create FileOperation = iota
+	Delete
+)
+
+// String() allows the FileOperation constants to be conveniently converted to a print-friendly format.
+func (fileOperation FileOperation) String() string {
+	return [...]string{"Create", "Delete"}[fileOperation]
+}
+
+// Intent stores the next intended file operation of Hermes.
+// This is used as part of the StateJournal of Hermes.
+type Intent struct {
```

Comment:
I dunno what format you're going to write these out in, but if it's a proto, then you probably want a `proto3` definition instead of a golang `struct`, and if it's `JSON` then you should add annotations like [this](https://gobyexample.com/json).

This is true here and elsewhere.

Reply:
I think that JSON makes sense here for writing this file out, but we can also go for a proto definition instead. I've implemented this change with JSON, but I can alter it to be proto2/3.

Reply:
Only use JSON when interacting another system that requires it.
When exchanging or persisting state, use binary protos (*not* text protos).

Reply:
Please change this from JSON to proto.

Reply:
I have changed this to proto in my latest commit.